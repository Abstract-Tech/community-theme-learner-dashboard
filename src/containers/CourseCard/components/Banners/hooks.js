import { useIntl } from '@edx/frontend-platform/i18n';
import { useCardValues } from 'hooks';
import { selectors } from 'data/redux';

const { cardData } = selectors;

export const useCourseBannerData = ({ courseNumber }) => ({
  courseData: useCardValues(courseNumber, {
    isVerified: cardData.isVerified,
    isCourseRunActive: cardData.isCourseRunActive,
    canUpgrade: cardData.canUpgrade,
    isAuditAccessExpired: cardData.isAuditAccessExpired,
    courseWebsite: cardData.courseWebsite,
  }),
  formatMessage: useIntl().formatMessage,
});

export default {
  useCourseBannerData,
};
