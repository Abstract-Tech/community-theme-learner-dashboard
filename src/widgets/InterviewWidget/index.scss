#interview-widget {
  margin-bottom: 1em;
}
