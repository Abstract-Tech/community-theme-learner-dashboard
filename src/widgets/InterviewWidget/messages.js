import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  interviewTitle: {
    id: 'InterviewSidebar.interviewTitle',
    defaultMessage: 'Help shape the future of edX',
    description: 'Interview card heading',
  },
  interviewPrompt: {
    id: 'InterviewSidebar.interviewPrompt',
    defaultMessage: 'Sign up for a 30-minute Q&A session with an edX research expert!',
    description: 'Prompt user to sign up',
  },
  signUpButton: {
    id: 'InterviewSidebar.signUpButton',
    defaultMessage: 'Sign up',
    description: 'Button sign up',
  },
  dismissButton: {
    id: 'InterviewSidebar.dismissButton',
    defaultMessage: 'No thanks',
    description: 'decline prompt to sign up',
  },
});

export default messages;
