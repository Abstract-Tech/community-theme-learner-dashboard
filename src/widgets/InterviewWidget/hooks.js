import React from 'react';
import { StrictDict } from 'utils';

import { getCookie, setCookie } from '../../utils/cookies';
import { configuration } from '../../config';

import * as module from './hooks';

export const state = StrictDict({
  hideWidget: (val) => React.useState(val), // eslint-disable-line
});

export const useDismissPanel = () => {
  const dismissCookieName = configuration.INTERVIEW_WIDGET_DISMISS_COOKIE_NAME;
  const [hideWidget, setHideWidget] = module.state.hideWidget(
    !!getCookie(dismissCookieName) || !configuration.INTERVIEW_WIDGET_SIGNUP_URL,
  );

  const handleDismiss = () => {
    setHideWidget(true);
    setCookie(dismissCookieName, true, 365);
  };

  return {
    hideWidget,
    handleDismiss,
  };
};

export default useDismissPanel;
