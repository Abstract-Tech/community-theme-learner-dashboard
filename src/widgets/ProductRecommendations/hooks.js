import { useState, useEffect } from 'react';
import { RequestStates, RequestKeys } from 'data/constants/requests';
import { StrictDict } from 'utils';
import { reduxHooks } from 'hooks';
import { SortKeys } from 'data/constants/app';
import { getCookie } from 'utils/cookies';
import { useWindowSize, breakpoints } from '@edx/paragon';
import api from './api';
import * as module from './hooks';

export const state = StrictDict({
  requestState: (val) => useState(val), // eslint-disable-line
  data: (val) => useState(val), // eslint-disable-line
});

export const useIsMobile = () => {
  const { width } = useWindowSize();
  return width < breakpoints.small.minWidth;
};

export const useShowRecommendationsFooter = () => {
  const hasAvailableDashboards = reduxHooks.useHasAvailableDashboards();
  const hasRequestCompleted = reduxHooks.useRequestIsCompleted(RequestKeys.initialize);
  const userAttributes = {
    is_mobile_user: module.useIsMobile(),
    lang_preference: global.navigator.languages.join(' '),
    is_enterprise_user: !!hasAvailableDashboards,
    location: getCookie('prod-edx-cf-loc'),
  };

  console.log("USer attributes here: ", userAttributes);

  // Hardcoded to not show until experiment related code is implemented
  return {
    shouldShowFooter: true,
    shouldLoadFooter: hasRequestCompleted && !hasAvailableDashboards,
  };
};

export const useMostRecentCourseRunKey = () => {
  const mostRecentCourseRunKey = reduxHooks.useCurrentCourseList({
    sortBy: SortKeys.enrolled,
    filters: [],
    pageSize: 0,
  }).visible[0]?.courseRun?.courseId;

  return mostRecentCourseRunKey;
};

export const useFetchRecommendations = (setRequestState, setData) => {
  const courseRunKey = module.useMostRecentCourseRunKey();

  useEffect(() => {
    let isMounted = true;

    const handleSuccess = (response) => {
      if (isMounted) {
        setData(response.data);
        setRequestState(RequestStates.completed);
      }
    };

    const handleError = () => {
      if (isMounted) {
        setRequestState(RequestStates.failed);
      }
    };

    if (courseRunKey) {
      api
        .fetchCrossProductRecommendations(courseRunKey)
        .then(handleSuccess)
        .catch(handleError);
    } else {
      api
        .fetchAmplitudeRecommendations()
        .then(handleSuccess)
        .catch(handleError);
    }
    return () => {
      isMounted = false;
    };
    /* eslint-disable */
  }, []);
};

export const useProductRecommendationsData = () => {
  const [requestState, setRequestState] = module.state.requestState(RequestStates.pending);
  const [data, setData] = module.state.data({});
  module.useFetchRecommendations(setRequestState, setData);

  return {
    productRecommendations: data,
    isLoading: requestState === RequestStates.pending,
    isLoaded: requestState === RequestStates.completed,
    hasFailed: requestState === RequestStates.failed 
  };
};

export default { useProductRecommendationsData, useShowRecommendationsFooter };
