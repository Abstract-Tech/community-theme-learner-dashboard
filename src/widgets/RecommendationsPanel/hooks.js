import React from 'react';

import { StrictDict } from 'utils';
import { RequestStates } from 'data/constants/requests';
import track from './track';

import * as module from './hooks';
import api from './api';

export const searchCourseEventName = 'learner_home.widget.search_course';

export const state = StrictDict({
  requestState: (val) => React.useState(val), // eslint-disable-line
  data: (val) => React.useState(val), // eslint-disable-line
});

export const useFetchCourses = (setRequestState, setData) => {
  React.useEffect(() => {
    let isMounted = true;
    api.fetchRecommendedCourses().then((response) => {
      if (isMounted) {
        setRequestState(RequestStates.completed);
        setData(response);
      }
    }).catch(() => {
      if (isMounted) {
        setRequestState(RequestStates.failed);
      }
    });
    return () => { isMounted = false; };
    /* eslint-disable */
  }, []);
};

export const useRecommendationPanelData = () => {
  const [requestState, setRequestState] = module.state.requestState(RequestStates.pending);
  const [data, setData] = module.state.data({});
  module.useFetchCourses(setRequestState, setData);
  const courses = data.data?.courses || [];
  const isPersonalizedRecommendation = data.data?.isPersonalizedRecommendation || false;
  return {
    courses,
    isLoaded: requestState === RequestStates.completed && courses.length > 0,
    isFailed: requestState === RequestStates.failed
      || (requestState === RequestStates.completed && courses.length === 0),
    isLoading: requestState === RequestStates.pending,
    trackFindCoursesClicked: track.findCoursesClicked,
  };
};

export default {
  useRecommendationPanelData,
};
