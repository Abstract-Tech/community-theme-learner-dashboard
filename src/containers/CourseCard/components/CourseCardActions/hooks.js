import { useDispatch } from 'react-redux';

import { Locked } from '@edx/paragon/icons';
import { useIntl } from '@edx/frontend-platform/i18n';

import { hooks as appHooks } from 'data/redux';
import messages from './messages';

export const useCardActionData = ({ cardId }) => {
  const { formatMessage } = useIntl();
  const dispatch = useDispatch();
  const {
    canUpgrade,
    coursewareAccess,
    hasStarted,
    isAudit,
    isAuditAccessExpired,
    isVerified,
  } = appHooks.useCardEnrollmentData(cardId);
  const { isArchived } = appHooks.useCardCourseRunData(cardId);
  const {
    canChange,
    hasSessions,
    isEntitlement,
    isExpired,
    isFulfilled,
  } = appHooks.useCardEntitlementData(cardId);
  const openSessionModal = appHooks.useUpdateSelectSessionModalCallback(dispatch, cardId);

  const hasAccess = coursewareAccess.isStaff || !(
    coursewareAccess.hasUnmetPrereqs
    || coursewareAccess.isTooEarly
    || isArchived);

  const selectSessionButton = {
    children: formatMessage(messages.selectSession),
    onClick: openSessionModal,
    disabled: !hasAccess || (!canChange || !hasSessions),
  };

  const viewCourseButton = {
    children: formatMessage(messages.viewCourse),
    disabled: !hasAccess || (isEntitlement && isExpired),
  };

  const beginCourseButton = {
    children: formatMessage(messages.beginCourse),
    disabled: !hasAccess,
  };

  const resumeButton = {
    children: formatMessage(messages.resume),
    disabled: isAudit && isAuditAccessExpired,
  };

  const upgradeButton = {
    iconBefore: Locked,
    variant: 'outline-primary',
    children: formatMessage(messages.upgrade),
    disabled: !canUpgrade,
  };

  let primary;
  if (isEntitlement) {
    primary = isFulfilled ? selectSessionButton : viewCourseButton;
  } else if (!hasStarted) {
    primary = beginCourseButton;
  } else if (isArchived) {
    primary = viewCourseButton;
  } else {
    primary = resumeButton;
  }

  const secondary = (isEntitlement || !isVerified) ? null : upgradeButton;
  return { primary, secondary };
};

export default useCardActionData;
