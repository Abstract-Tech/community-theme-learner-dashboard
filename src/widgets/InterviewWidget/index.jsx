import { React } from 'react';

import { useIntl } from '@edx/frontend-platform/i18n';
import { Button, Card } from '@edx/paragon';
import { Calendar } from '@edx/paragon/icons';

import interviewWidgetImg from 'assets/interview-widget.png';

import { configuration } from '../../config';

import './index.scss';
import messages from './messages';
import { useDismissPanel } from './hooks';

export const InterviewWidget = () => {
  const {
    hideWidget,
    handleDismiss,
  } = useDismissPanel();
  const { formatMessage } = useIntl();

  if (hideWidget) {
    return null;
  }
  return (
    <Card id="interview-widget">
      <Card.ImageCap
        src={interviewWidgetImg}
        srcAlt="course side widget"
      />
      <Card.Header
        title={formatMessage(messages.interviewTitle)}
      />
      <Card.Section>
        {formatMessage(messages.interviewPrompt)}
      </Card.Section>
      <Card.Footer>
        <Button variant="tertiary" onClick={handleDismiss}>{formatMessage(messages.dismissButton)}</Button>
        <Button
          variant="outline-primary"
          iconBefore={Calendar}
          href={configuration.INTERVIEW_WIDGET_SIGNUP_URL}
        >
          {formatMessage(messages.signUpButton)}
        </Button>
      </Card.Footer>
    </Card>
  );
};

export default InterviewWidget;
