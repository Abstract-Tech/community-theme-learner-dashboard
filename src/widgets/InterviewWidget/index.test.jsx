import { shallow } from 'enzyme';

import * as hooks from './hooks';
import InterviewWidget from '.';

jest.mock('./hooks', () => ({
  useDismissPanel: jest.fn(),
}));

describe('InterviewWidget', () => {
  describe('snapshots', () => {
    test('default', () => {
      hooks.useDismissPanel.mockReturnValueOnce({
        hideWidget: false,
        handleDismiss: jest.fn(),
      });
      const wrapper = shallow(<InterviewWidget />);
      expect(wrapper).toMatchSnapshot();
    });
    test('dismissed', () => {
      hooks.useDismissPanel.mockReturnValueOnce({
        hideWidget: true,
        handleDismiss: jest.fn(),
      });
      const wrapper = shallow(<InterviewWidget />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
