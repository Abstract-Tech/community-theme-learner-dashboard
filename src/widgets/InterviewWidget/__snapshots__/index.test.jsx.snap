// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterviewWidget snapshots default 1`] = `
<Card
  id="interview-widget"
>
  <Card.ImageCap
    src="test-file-stub"
    srcAlt="course side widget"
  />
  <Card.Header
    title="Help shape the future of edX"
  />
  <Card.Section>
    Sign up for a 30-minute Q&A session with an edX research expert!
  </Card.Section>
  <Card.Footer>
    <Button
      onClick={[MockFunction]}
      variant="tertiary"
    >
      No thanks
    </Button>
    <Button
      href="https://test.example.com"
      variant="outline-primary"
    >
      Sign up
    </Button>
  </Card.Footer>
</Card>
`;

exports[`InterviewWidget snapshots dismissed 1`] = `""`;
